types/coin.go

```diff
@@ -842,45 +841,16 @@ func (coins Coins) Sort() Coins {
 	return coins
 }
 
-//-----------------------------------------------------------------------------
-// Parsing
-
-var (
-	// Denominations can be 3 ~ 128 characters long and support letters, followed by either
-	// a letter, a number or a separator ('/', ':', '.', '_' or '-').
-	reDnmString = `[a-zA-Z][a-zA-Z0-9/:._-]{2,127}`
-	reDecAmt    = `[[:digit:]]+(?:\.[[:digit:]]+)?|\.[[:digit:]]+`
-	reSpc       = `[[:space:]]*`
-	reDnm       *regexp.Regexp
-	reDecCoin   *regexp.Regexp
-)
-
-func init() {
-	SetCoinDenomRegex(DefaultCoinDenomRegex)
-}
-
-// DefaultCoinDenomRegex returns the default regex string
-func DefaultCoinDenomRegex() string {
-	return reDnmString
-}
-
-// coinDenomRegex returns the current regex string and can be overwritten for custom validation
-var coinDenomRegex = DefaultCoinDenomRegex
-
-// SetCoinDenomRegex allows for coin's custom validation by overriding the regular
-// expression string used for denom validation.
-func SetCoinDenomRegex(reFn func() string) {
```

Comment:
> Maybe change this to a custom parsing fn instead of custom regex?

@kocubinski apologies if I'm missing something, wdym by changing this to a custom parsing func? 👀

Reply:
it makes sense to keep the same logic here, the default will be the regel generated code. Then the current feature of allowing teams to use custom regex would still work.

Reply:
Understood. Done :)